extmod/modssl_mbedtls.c

```diff
@@ -264,6 +409,73 @@ STATIC void ssl_context_load_key(mp_obj_ssl_context_t *self, mp_obj_t key_obj, m
     }
 }
 
+#ifdef MICROPY_SSL_MBEDTLS_EXTRAS
+
+STATIC mp_obj_t mod_ssl_load_certchain(size_t n_args, const mp_obj_t *pos_args,
+    mp_map_t *kw_args) {
+
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_keyfile, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = MP_ROM_NONE} },
+    };
+
+    mp_obj_ssl_context_t *self = MP_OBJ_TO_PTR(pos_args[0]);
+    mp_obj_t pkey;
+    mp_obj_t cert;
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args - 2, pos_args + 2, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    if (pos_args[1] != mp_const_none) {
+        mp_check_self(mp_obj_is_str_or_bytes(pos_args[1])); // CERT
+        mp_check_self(mp_obj_is_str_or_bytes(args[0].u_obj)); // KEY
```

Comment:
These `mp_check_self()` calls are not needed.  The type will be checked automatically when the str/bytes value is extracted.